### PR TITLE
Merge up 1 6 to master

### PR DESCRIPTION
**Pull request by steveax**

### CODE REVIEW

File: lib/src/protocol/v1/serialization.cc [hunk 1]
```diff
@@ -9,7 +9,7 @@
 namespace PCPClient {
 namespace v1 {
 
-#ifdef BOOST_ENDIAN_LITTLE_BYTE
+#if BOOST_ENDIAN_LITTLE_BYTE
```
Comment (donoghuc) [inline lib/src/protocol/v1/serialization.cc:12]: This looks wrong. https://github.com/puppetlabs/cpp-pcp-client/commit/741ee6897088607737f28acce2d83592e65d7a00

**Comment by donoghuc** [comment]
Once travis/appveyor are done we should ask jenkins to test this with pxp-agent. 
```
jenkins please test pxp-agent
```

File: lib/src/protocol/v1/serialization.cc (hunk 1 above)
Comment (steveax) [reply lib/src/protocol/v1/serialization.cc:12]: yet, April 12: https://github.com/puppetlabs/cpp-pcp-client/commit/36ec2eba1ee6c0edeaaaaad9836041f84a02e463#diff-03a0c044596526965eba33d65a5657dbR12

Comment (steveax) [reply lib/src/protocol/v1/serialization.cc:12]: 🤷

Comment (MikaelSmith) [reply lib/src/protocol/v1/serialization.cc:12]: Yes, it does look wrong. Not sure how `#if...#endif` even compiles.

Comment (steveax) [reply lib/src/protocol/v1/serialization.cc]: Updated to go with what was in master. Should that change in 1.6.x too?

Comment (donoghuc) [reply lib/src/protocol/v1/serialization.cc]: Yes. I missed that on my review for the PR that was just merged. https://github.com/puppetlabs/cpp-pcp-client/pull/254 my bad

**Comment by donoghuc** [comment]
jenkins please test pxp-agent

File: lib/src/protocol/v1/serialization.cc (hunk 1 above)
Comment (steveax) [reply lib/src/protocol/v1/serialization.cc]: #256

**Pull request merged**